### PR TITLE
feat: cli oauth login

### PR DESCRIPTION
**Pull request by ndom91**
I hereby confirm that I followed the code guidelines found at [engineering guidelines](https://www.notion.so/checkly/Engineering-Guidelines-a3a165a203a04dc1a112f0e26b2f2d3f)

## ⚙️ Affected Components
* [x] Commands
* [ ] Modules
* [ ] Services
* [ ] SDK

### 📝 Notes for the Reviewer

Add oauth login to get user based api token.

Initial OAuth login works, but returns the account based API Key for now. 

image

cli_login

Next steps are to implement the user based apiKeys in the backend.

Backend Sister PR: https://github.com/checkly/checkly-backend/pull/2199

### CODE REVIEW

File: src/services/login-util.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    // `req.url` has a '/' char at the beginning which needs removed to be valid searchParams input
+    const responseParams = new URLSearchParams(req.url.substring(1))
+    const code = responseParams.get('code')
+    const state = responseParams.get('state')
+
+    if (code && state) {
+      res.write(`
+      <html>
+      <body>
+        <div style="height:100%;width:100%;inset:0;position:absolute;display:grid;place-items:center;background-color:#EFF2F7;text-align:center;font-family:Inter;">
```
Comment (umutuzgur) [inline src/services/login-util.js:60]: As we discussed, I think the Vercel cli directly exits when the auth is successful

File: src/services/login-util.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    .replace(/\//g, '_')
+    .replace(/=+$/, '')
+
+  return {
+    codeChallenge,
+    codeVerifier,
+  }
+}
+
+function startServer(getToken) {
```
Comment (umutuzgur) [inline src/services/login-util.js:48]: I would rename the getToken to something more callbacky. Like, onTokenCallback

**Review by umutuzgur — CHANGES_REQUESTED** [review]
I have just one naming change request, the rest look good and I will approve it afterwards

File: src/services/login-util.js (hunk 2 above)
Comment (ndom91) [reply src/services/login-util.js:48]: Nice, dig it. Much clearer :+1:

File: src/services/login-util.js (hunk 1 above)
Comment (ndom91) [reply src/services/login-util.js:60]: Ah yes, implemented! :+1:

Comment (umutuzgur) [reply src/services/login-util.js:60]: nice

**Review by umutuzgur — APPROVED** [review]
👌

**Pull request merged**